File: typescript/relational-database-mapping/src/soccer/soccer.ts
import {Athlete} from "../athlete/athlete.ts";

export class Soccer extends Athlete {

  public readonly club: string;

  constructor(name: string, club: string, type: string) {
    super(name, type);
    this.club = club;
  }

  getClub(): string {
    return this.club;
  }
}

File: java/relational-database-mapping/src/main/java/soccer/Soccer.java
package soccer;

import athlete.Athlete;

public class Soccer extends Athlete {

    private final String club;

    public Soccer(final String name, final String type, final String club) {
        super(name, type);
        this.club = club;
    }

    public final String getClub() {
        return club;
    }
}
